File: July/validTriangleNumber.js
const triangleNumber = (nums = []) => {
    nums.sort((a, b) => a - b)
    let k = 2
    let count = 0
    for (let i = 0; i < nums.length - 2; i++) {
      for (let j = i + 1; j < nums.length - 1; j++) {
        k = j + 1
        while (nums[k] < nums[i] + nums[j]) {
          k += 1
        }
        count += k - j - 1
      }
    }
    return count
}

File: July/validTriangleNumber.py
class Solution:
    def triangleNumber(self, nums):
        n=len(nums)
        if n<3:
            return 0
        nums=sorted(nums)
        nums=nums[::-1]
        count=0
        for i in range(0,n-2):
            left=i+1
            right=n-1
            while left<right:
                if nums[left]+nums[right]>nums[i]:
                    count+=(right-left)
                    left+=1
                else:
                    right-=1
        return count

